### PR TITLE
Don't remove <head> contents during sanitization.

### PR DESCRIPTION
**Pull request by emilio**
The DOMPurify preserves `<base>` elements, and they're handled by
transformBase.

But if the `<base>` element is in the `<head>`, it gets accidentally trimmed
too early via `cheerio('body').html()`, which just discards `<head>`
contents.

Fix it by using `cheerio.html()` which returns the complete html of the
document.

### CODE REVIEW

File: packages/shared/lib/sanitize/purify.ts [hunk 1]
```diff
@@ -27,7 +27,7 @@ const CONFIG: { [key: string]: any } = {
     raw: { WHOLE_DOCUMENT: true, RETURN_DOM: true },
     html: { WHOLE_DOCUMENT: false, RETURN_DOM: true },
     protonizer: {
-        FORBID_TAGS: ['input', 'form'], // Overide defaults to allow style (will be processed by juice afterward)
+        FORBID_TAGS: ['input', 'form'], // Override defaults to allow style (will be processed by juice afterward)
```
Comment (emilio) [inline packages/shared/lib/sanitize/purify.ts:30]: Let me know if you want this typo fix to be in a separate PR or something.

**Comment by Swiip** [comment]
Thanks a lot, looks good to me, looks like I introduced a regression when I updated the cheerio code.

Yet I still need to double check there is no further implication about letting the head pass that phase. There could be a lot more things than a base tag in a head.

I'll try to let you know when it's done.

**Comment by emilio** [comment]
> Yet I still need to double check there is no further implication about letting the head pass that phase. There could be a lot more things than a base tag in a head.

Given any element in `<head>` can effectively also be under the `<body>` element, I don't think there should be any new implication in practice, right?

**Comment by mmso** [comment]
Merged in https://github.com/ProtonMail/WebClients/commit/661185c1c9855688e7f73adb08afa3ed4ed5eb43. Thanks for your contribution!
